vm/src/main/scala/pravda/vm/standard/ExponentialFunction.scala

```diff
@@ -0,0 +1,317 @@
+    }
+    implicit class LongExp(x: Long) {
+      def **(y: Double): Long = math.pow(x.toDouble, y).toLong
+      def **(y: Byte): Long = math.pow(x.toDouble, y.toDouble).toLong
+      def **(y: Int): Long = math.pow(x.toDouble, y.toDouble).toLong
+      def **(y: Long): Long = math.pow(x.toDouble, y.toDouble).toLong
+      def **(y: Short): Long = math.pow(x.toDouble, y.toDouble).toLong
+    }
+    implicit class IntExp(x: Int) {
+      def **(y: Double): Int = math.pow(x.toDouble, y).toInt
+      def **(y: Byte): Int = math.pow(x.toDouble, y.toDouble).toInt
+      def **(y: Int): Int = math.pow(x.toDouble, y.toDouble).toInt
+      def **(y: Long): Int = math.pow(x.toDouble, y.toDouble).toInt
+      def **(y: Short): Int = math.pow(x.toDouble, y.toDouble).toInt
+    }
+    implicit class ShortExp(x: Short) {
+      def **(y: Double): Short = math.pow(x.toDouble, y).toShort
+      def **(y: Byte): Short = math.pow(x.toDouble, y.toDouble).toShort
+      def **(y: Int): Short = math.pow(x.toDouble, y.toDouble).toShort
+      def **(y: Long): Short = math.pow(x.toDouble, y.toDouble).toShort
+      def **(y: Short): Short = math.pow(x.toDouble, y.toDouble).toShort
+    }
+    implicit class ByteExp(x: Byte) {
+      def **(y: Double): Int = math.pow(x.toDouble, y).toInt
+      def **(y: Byte): Int = math.pow(x.toDouble, y.toDouble).toInt
+      def **(y: Int): Int = math.pow(x.toDouble, y.toDouble).toInt
+      def **(y: Long): Int = math.pow(x.toDouble, y.toDouble).toInt
+      def **(y: Short): Int = math.pow(x.toDouble, y.toDouble).toInt
+    }
+
+    def getCost(x: Double, y: Double): Long = {
+      (math.log10(math.abs(x) + 1) / div * math.abs(y) / 8).toLong + 1
+    }
+
+    def getBigIntCost(x: scala.BigInt, y: Int): Long = {
+      (x.abs.toString().length / div * math.abs(y) / 8).toLong + 1
+    }
+    wattCounter.cpuUsage(CpuArithmetic)
+    val a = memory.pop()
+    val b = memory.pop()
+
+    wattCounter.cpuUsage(CpuArithmetic)
+
+    val res = a match {
+      case Int32(lhs) =>
+        b match {
+          case Int8(rhs) =>
+            wattCounter.memoryUsage(getCost(lhs.toDouble, rhs.toDouble))
+            Int32(lhs ** rhs)
+          case Int16(rhs) =>
+            wattCounter.memoryUsage(getCost(lhs.toDouble, rhs.toDouble))
+            Int32(lhs ** rhs)
+          case Int32(rhs) =>
+            wattCounter.memoryUsage(getCost(lhs.toDouble, rhs.toDouble))
+
+            Int32(lhs ** rhs)
+          case Uint8(rhs) =>
+            wattCounter.memoryUsage(getCost(lhs.toDouble, rhs.toDouble))
+            Int32(lhs ** rhs)
+          case Uint16(rhs) =>
+            wattCounter.memoryUsage(getCost(lhs.toDouble, rhs.toDouble))
+            Int32(lhs ** rhs)
+          case Uint32(rhs) =>
+            wattCounter.memoryUsage(getCost(lhs.toDouble, rhs.toDouble))
+            Int32(lhs ** rhs)
+          case BigInt(rhs) =>
+            wattCounter.memoryUsage(getCost(lhs.toDouble, rhs.doubleValue()))
```

Comment:
It's dangerous to convert `BigInt` to `Double`. It can be evaluated to `POSITIVE_INFINITY`. Write a method that takes two `BigInt`s and then convert the other argument to `BigInt` here.